src/utils/setupWriteToDisk.js

```diff
@@ -21,6 +21,20 @@ function setupWriteToDisk(context) {
     (context.compiler).compilers || [context.compiler];
 
   for (const compiler of compilers) {
+    if (compiler.outputFileSystem && compiler.options.output.clean) {
+      /** @type {Compiler["outputFileSystem"]} */
+      const { unlink: originalUnlink } = compiler.outputFileSystem;
+      if (originalUnlink) {
+        compiler.outputFileSystem.unlink = (
+          /** @type {String} */ originalPath,
+          /** @type {(arg0?: null | NodeJS.ErrnoException) => void} */ originalCallback
+        ) => {
+          fs.unlink(originalPath, () => {});
```

Comment:
Hello, there is racing you should when the first operation finished,  also using `fs.unlink` is not safe, because some developer use custom `outputFileSystem` to avoid file removing, but we remove them and break application, we nee to think deeply how we can handle such case. In theiry we can apply clean plugin before watch and it will with original fs